src/stdatamodels/jwst/_kwtool/compare.py

```diff
@@ -65,6 +65,20 @@ def __repr__(self):
             },
         },
     },
+    ("PRIMARY", "FOCUSPOS"): {
+        "type": {
+            "dmd": {
+                "union": {"integer"},
```

Comment:
FOCUSPOS for nirspec is listed as "steps" and of type "integer". For niriss and fg it's listed as "mm" and of type "float".

Adding this expected difference allows it to be listed as a "number" in the datamodel schema.